### PR TITLE
Add JSDoc for extras/splat ply-reader/splat-material/splat

### PR DESCRIPTION
**Pull request by kungfooman**
Also adding `jsconfig.js` and fixing a small issue with `find` for `UInt8Array`'s (it would skip a good/last result) because it used `<` and not `<=`.

Motivation is to have a better IDE experience (static type-checking and Intellisense) and also to have more types in playcanvas-extras that we can validate via RTI (dynamic type-checking for NaN's for example).

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by kungfooman** [comment]
I think I found another bug:

image

The property `app.loader.maxRetries` doesn't exist, but they do exist on each `containerHandler`, can I just fix that here, like `containerHandler.maxRetries`?

**Comment by mvaligursky** [comment]
This might be related
https://github.com/playcanvas/engine/pull/5574

File: extras/splat/ply-reader.js [hunk 1]
```diff
@@ -12,25 +12,32 @@ const dataTypeMap = new Map([
...
-    properties;
-}
-
-// asynchronously read a ply file data
+/**
+ * @typedef {Int8Array|Uint8Array|Int16Array|Uint16Array|Int32Array|Uint32Array|Float32Array|Float64Array} DataType
+ */
+
+/**
+ * @typedef {object} PlyProperty
```
Comment (mvaligursky) [inline extras/splat/ply-reader.js:20]: do we even use these types anywhere?

Comment (kungfooman) [reply extras/splat/ply-reader.js:20]: yep, `PlyElement` is used once in `ply-reader.js` and three times in `splat-data.js`:

```js
class SplatData {
    /** @type {import('./ply-reader').PlyElement[]} */
    elements;

    /** @type {import('./ply-reader').PlyElement} */
    vertexElement;

    /**
     * @param {import('./ply-reader').PlyElement[]} elements - The elements.
     * @param {boolean} [performZScale] - Whether to perform z scaling.
     */
    constructor(elements, performZScale = true) {
```

**Review by slimbuck — APPROVED** [review]
Yep, looks great thanks!

**Pull request merged**